File: lib/features/scanner/domain/product.dart
class Product {
  final String id;
  final String name;
  final double price;
  final String qrCode;
  final String imageUrl;
  final String category;

  Product({
    required this.id,
    required this.name,
    required this.price,
    required this.qrCode,
    required this.imageUrl,
    required this.category,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      price: (json['price'] ?? 0.0).toDouble(),
      qrCode: json['qr_code'] ?? '',
      imageUrl: json['image_url'] ?? '',
      category: json['category'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'price': price,
      'qr_code': qrCode,
      'image_url': imageUrl,
      'category': category,
    };
  }
}

File: backend/src/models/Product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
    storeId: {
        type: String,
        required: [true, 'Store ID is required to distinguish products between Dmart, Zudio, etc.'],
        index: true,
        enum: ['dmart', 'zudio'], // Enforce our two stores
        lowercase: true,
    },
    qr_code: {
        type: String,
        required: [true, 'QR Code is required'],
        index: true,
    },
    name: {
        type: String,
        required: [true, 'Product name is required'],
    },
    price: {
        type: Number,
        required: [true, 'Product price is required'],
        min: 0,
    },
    imageUrl: {
        type: String,
        default: '',
    },
    category: {
        type: String,
        default: 'General',
    },
}, { timestamps: true });

// A qr_code should be unique ONLY within its specific store
productSchema.index({ storeId: 1, qr_code: 1 }, { unique: true });

const Product = mongoose.model('Product', productSchema);
module.exports = Product;
